src/plugins/home/public/application/components/solutions_section/solution_title.tsx

```diff
@@ -42,24 +42,107 @@ import {
 } from '@elastic/eui';
 
 interface Props {
-  title: string;
   subtitle: string;
   iconType: IconType;
+  branding: Branding;
 }
 
-export const SolutionTitle: FC<Props> = ({ title, subtitle, iconType }) => (
+/**
+ * @param {boolean} darkMode
+ * @param {object} mark - defaultUrl will be used in default mode; darkModeUrl will be used in dark mode
+ * @param {string} applicationTitle - custom title for the application
+ */
+export interface Branding {
+  darkMode: boolean;
+  mark: {
+    defaultUrl?: string;
+    darkModeUrl?: string;
+  };
+  applicationTitle?: string;
+}
+
+/**
+ * Use branding configurations to check which URL to use for rendering
+ * home card logo in default mode. In default mode, home card logo will
+ * proritize default mode mark URL. If it is invalid, default opensearch logo
+ * will be rendered.
+ *
+ * @returns a valid custom URL or undefined if no valid URL is provided
+ */
+const customHomeLogoDefaultMode = (branding: Branding) => {
+  return branding.mark.defaultUrl ?? undefined;
+};
+
+/**
+ * Use branding configurations to check which URL to use for rendering
+ * home logo in dark mode. In dark mode, home logo will render
+ * dark mode mark URL if valid. Otherwise, it will render the default
+ * mode mark URL if valid. If both dark mode mark URL and default mode mark
+ * URL are invalid, the default opensearch logo will be rendered.
+ *
+ * @returns {string|undefined} a valid custom URL or undefined if no valid URL is provided
+ */
+const customHomeLogoDarkMode = (branding: Branding) => {
+  return branding.mark.darkModeUrl ?? branding.mark.defaultUrl ?? undefined;
+};
+
+/**
+ * Render custom home logo for both default mode and dark mode
+ *
+ * @returns {string|undefined} a valid custom loading logo URL, or undefined
+ */
+const customHomeLogo = (branding: Branding) => {
+  if (branding.darkMode) {
+    return customHomeLogoDarkMode(branding);
+  }
+  return customHomeLogoDefaultMode(branding);
+};
+
+/**
+ * Check if we render a custom home logo or the default opensearch spinner.
+ * If customWelcomeLogo() returns undefined(no valid custom URL is found), we
+ * render the default opensearch logo
+ *
+ * @returns a image component with custom logo URL, or the default opensearch logo
+ */
+const renderBrandingEnabledOrDisabledLogo = (iconType: IconType, branding: Branding) => {
+  const customLogo = customHomeLogo(branding);
+  if (customLogo) {
```

Comment:
Will this logic ever be the case? Essentially do we want to replace iconType or try to deprecate it in favor the new param branding?

Reply:
Yes, we can deprecate the original Prop iconType and title in favor of the new branding param.

Reply:
Also replaced the inputOutput door logo with official opensearch mark.